RunningOrder/Shared/Components/Search/SearchBarSuggestions+Logic.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  SearchBarSuggestions+Logic.swift
+//  RunningOrder
+//
+//  Created by Ghita Laoud on 26/03/2021.
+//  Copyright © 2021 Worldline. All rights reserved.
+//
+
+import Foundation
+import Combine
+import SwiftUI
+
+extension SearchBarSuggestions {
+    final class Logic: ObservableObject {
+        @Binding var input: String
+        private unowned var storyManager: StoryManager
+        private unowned var sprintManager: SprintManager
+        var searchManager: SearchManager
+
+        init(input: Binding<String>, storyManager: StoryManager, sprintManager: SprintManager, searchManager: SearchManager) {
+            self._input = input
+            self.storyManager = storyManager
+            self.sprintManager = sprintManager
+            self.searchManager = searchManager
+        }
+
+        /// Return filtered stories
+        /// - Parameter filter: user's input
+        /// - Returns: All stories in all sprints filtered
+        var filteredStories: [Story] {
+            var wholeStories: [Story] = []
+            sprintManager.sprints.forEach { sprint in
+                let storiesPerId = storyManager.stories[sprint.id] ?? []
+                wholeStories.append(contentsOf: storiesPerId)
+            }
+
+            return wholeStories.filter { story -> Bool in
+                return story.name.lowercased().contains(input.lowercased()) || story.epic.lowercased().contains(input.lowercased()) || story.ticketReference.lowercased().contains(input.lowercased())
+            }
```

Comment:
```suggestion
            sprintManager.sprints
               .flatMap { sprint in return storyManager.stories[sprint.id] ?? [] }
               .filter { story -> Bool in
                   return story.name.lowercased().contains(input.lowercased()) || story.epic.lowercased().contains(input.lowercased()) || story.ticketReference.lowercased().contains(input.lowercased())
            }
```

could something like this be better ?